co-authors-plus.php

```diff
@@ -1052,22 +1052,59 @@ function filter_count_user_posts( $count, $user_id ) {
 
 		if ( is_object( $user ) ) {
 			$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
-
 			$term = $this->get_author_term( $user );
 
-			if ( $term && ! is_wp_error( $term ) ) {
-				if ( 'guest-author' === $user->type ) {
-					// If using guest author term count, add on linked user count.
-					$count = (int) $count + $term->count;
-				} else {
-					$count = $term->count;
-				}
+			if ( ! $user->linked_account ) {
+				$count = $term->count;
+			} else {
+				$count = $this->get_linked_post_count( $user->user_nicename, $user->linked_account );
 			}
 		}
 
 		return $count;
 	}
 
+	/**
+	 * Get the post count of linked accounts.
+	 *
+	 * @param string $user_account The name of the user account.
+	 * @param string $linked_account The name of the linked account.
+	 *
+	 * @return int The post count of linked accounts.
+	 */
+	function get_linked_post_count( $user_account, $linked_account ) {
+		$user_account_ids   = $this->get_post_ids_by_author_name( $user_account );
+		$linked_account_ids = $this->get_post_ids_by_author_name( $linked_account );
+
+		return count( array_unique( array_merge( $user_account_ids, $linked_account_ids ) ) );
+	}
+
+	/**
+	 * Get the post IDs by author name.
+	 *
+	 * @param string $author_name The name of the corresponding user.
+	 *
+	 * @return array The IDs of all matching posts.
+	 */
+	function get_post_ids_by_author_name( $author_name ) {
```

Comment:
Good catch, @htdat !  

From my looking at this issue - is it a problem we can solve with the current code without needing to add more queries or functions?

Reply:
> I am concerned that this may be a performance and accuracy issue if either accounts or both have so many posts. Let's think of a few thousand posts at least.

> There is a limit of how many rows can be returned for each SQL command on VIP Go. If I remember correctly, it's 1000.

You're right, @htdat. I haven't thought about that while developing this fix. I think it's best to process the query in batches.

> From my looking at this issue - is it a problem we can solve with the current code without needing to add more queries or functions?

I'm afraid that's not possible, @mikeyarce. The default WordPress functionality does not consider having multiple authors for one post. Let's say you and I are working on the same post and our accounts get linked. Running `count_user_posts()` would result in 2 instead of in 1, as both of us are listed as author in the corresponding post.

That said, if you have an idea on how to solve this problem without adding extra code, I'm all ears.

Reply:
> The default WordPress functionality does not consider having multiple authors for one post

Ahhh interesting, is this another bug?  or is this something caught by the current unit tests?

One thought I had was - in this case, is the bug the counting or the fact that you have 1 author added to a post twice?   Maybe the count is right but the plugins shouldn't keep both of those authors on the post if they are linked?

Reply:
> One thought I had was - in this case, is the bug the counting or the fact that you have 1 author added to a post twice?

I wouldn't say that this is a bug as by default a page or post can only have one authors. The `count_user_posts()` function does simply not consider that a page of post can have multiple authors. 

> Maybe the count is right but the plugins shouldn't keep both of those authors on the post if they are linked?

Imagine you create the post `Performant WordPress Queries` with the guest account `VIP Developer` and your regular account `Mikey`. Running `count_user_posts()` shows a post count of 1 for both accounts. You're now linking the guest account with your regular account. `count_user_posts()` still returns 1 for both accounts. For linked accounts, the individual number of all pages and posts will be cumulated. This results in a 2, which is incorrect as both accounts were involved in the same post and not in two different posts.

I'd say, this is a very specific problem that is created by the CAP plugin and therefore needs custom code to calculate the proper post count.

Reply:
> I think it's necessary to have unit tests for the 3 mentioned cases here.

@htdat I just noticed that I haven't replied to that suggestion yet. I'll definitely create unit tests for this issue. Especially given the fact that there are several attempts to resolve this particular problem, unit tests should be added to ensure that future developments do not break this fix.

Reply:
> That said, if you have an idea on how to solve this problem without adding extra code, I'm all ears.

@nielslange  - What's about if approaching this issue a bit differently? Instead of getting all post IDs from the WP user and the guest author and comparing them, we do this math: 

- (1) Total posts of the WP User
- (2) Total posts of the guest author.
- (3) Total shared posts between this WP User and this guest author. 

The final result is: (1) + (2) - (3). 

To calculate (3), I can think of a count query to get all posts from this specific real user ID having the guest account term. The only concern is that if there is any performant issue with this SQL query.

Reply:
That's a neat idea, @htdat. I'll try that right now.

Reply:
If we can use counts over introducing additional queries, that would be ideal.

Reply:
If we can use counts over introducing additional queries, that would be ideal. The WP_Term class has the `count` property, have you had a chance to look into utilizing that? https://developer.wordpress.org/reference/classes/wp_term/.

Reply:
@rebeccahum After looking into @htdat's suggestion, I rewrote the function and I'm using `COUNT` now.

@htdat Based on your suggestion, I created a query that returns the combined post count directly. Thus, I'm doing (1), (2) and (3) directly in the query.

Reply:
@nielslange Is there a reason why you left this function since it's not being used anymore?

Reply:
I missed removing that function initially. I just removed it and pushed an update.